src/cms/utils/pdf_utils.py

```diff
@@ -0,0 +1,142 @@
+    :rtype: ~django.http.HttpResponse
+    """
+    # first all necessary data for hashing are collected, starting at region_slug
+    pdf_key_list = [region_slug]
+    for page in pages:
+        # add translation id and last_updated to hash key list if they exist
+        page_translation = page.get_public_translation(language_code)
+        if page_translation:
+            # if translation for this language exists
+            pdf_key_list.append(page_translation.id)
+            pdf_key_list.append(page_translation.last_updated)
+        else:
+            # if the page has no translation for this language
+            pages = pages.exclude(id=page.id)
+    # finally combine all list entries to a single hash key
+    pdf_key_string = "_".join(map(str, pdf_key_list))
+    # compute the hash value based on the hash key
+    pdf_hash = hashlib.sha256(bytes(pdf_key_string, "utf-8")).hexdigest()
+    cached_response = cache.get(pdf_hash, "has_expired")
+    if cached_response != "has_expired":
+        # if django cache already contains a response object
+        return cached_response
+
+    amount_pages = pages.count()
+    language = Language.objects.get(code=language_code)
+    context = {
+        "html2pdf": True,
+        "right_to_left": language.text_direction == text_directions.RIGHT_TO_LEFT,
+        "region": region,
+        "pages": pages,
+        "language": language,
+        "amount_pages": amount_pages,
+    }
+    response = HttpResponse(content_type="application/pdf")
+    if amount_pages == 0:
+        return HttpResponse(_("No valid pages selected for PDF generation."))
+    if amount_pages == 1:
+        # If pdf contains only one page, take its title as filename
+        title = pages.first().get_public_translation(language_code).title
+    else:
+        # If pdf contains multiple pages, check the minimum level
+        min_level = pages.aggregate(Min("level")).get("level__min")
+        # Query all pages with this minimum level
+        min_level_pages = pages.filter(level=min_level)
+        if min_level_pages.count() == 1:
+            # If there's exactly one page with the minimum level, take its title
+            title = min_level_pages.first().get_public_translation(language_code).title
+        else:
+            # In any other case, take the region name
+            title = region.name
+    filename = f"Integreat - {language.translated_name} - {title}.pdf"
+    response["Content-Disposition"] = f'filename="{filename}"'
+    html = get_template("pages/page_pdf.html").render(context)
+    pisa_status = pisa.CreatePDF(
+        html, dest=response, link_callback=link_callback, encoding="UTF-8"
+    )
+    if pisa_status.err:
+        logger.error(
+            "The following PDF could not be rendered: Region: %s, Language: %s, Pages: %s.",
+            region,
+            language,
+            pages,
+        )
+        return HttpResponse(_("The PDF could not be successfully generated."))
```

Comment:
```suggestion
        return HttpResponse(_("The PDF could not be successfully generated."), status=500)
```
Use `500 Internal Server Error` to help the client understand the problem.